cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaMirrorMaker2Cluster.java

```diff
@@ -0,0 +1,316 @@
+        final StringBuilder clustersOauthAccessTokens = new StringBuilder();
+        final StringBuilder clustersOauthRefreshTokens = new StringBuilder();
+
+        for (KafkaMirrorMaker2ClusterSpec mirrorMaker2Cluster : clusters) {
+            String clusterAlias = mirrorMaker2Cluster.getAlias();
+
+            if (clusterAliases.length() > 0) {
+                clusterAliases.append(";");
+            }
+            clusterAliases.append(clusterAlias);
+
+            getClusterTrustedCerts(clustersTrustedCerts, mirrorMaker2Cluster, clusterAlias);
+
+            KafkaClientAuthentication authentication = mirrorMaker2Cluster.getAuthentication();
+            if (authentication != null) {
+                if (authentication instanceof KafkaClientAuthenticationTls) {
+                    KafkaClientAuthenticationTls tlsAuth = (KafkaClientAuthenticationTls) authentication;
+                    if (tlsAuth.getCertificateAndKey() != null) {
+                        appendCluster(clustersTlsAuthCerts, clusterAlias, () -> tlsAuth.getCertificateAndKey().getSecretName() + "/" + tlsAuth.getCertificateAndKey().getCertificate());
+                        appendCluster(clustersTlsAuthKeys, clusterAlias, () -> tlsAuth.getCertificateAndKey().getSecretName() + "/" + tlsAuth.getCertificateAndKey().getKey());
+                    }
+                } else if (authentication instanceof KafkaClientAuthenticationPlain) {
+                    KafkaClientAuthenticationPlain passwordAuth = (KafkaClientAuthenticationPlain) authentication;
+                    appendClusterPasswordSecretSource(clustersSaslPasswordFiles, clusterAlias, passwordAuth.getPasswordSecret());
+                } else if (authentication instanceof KafkaClientAuthenticationScramSha512) {
+                    KafkaClientAuthenticationScramSha512 passwordAuth = (KafkaClientAuthenticationScramSha512) authentication;
+                    appendClusterPasswordSecretSource(clustersSaslPasswordFiles, clusterAlias, passwordAuth.getPasswordSecret());
+                } else if (authentication instanceof KafkaClientAuthenticationOAuth) {
+                    KafkaClientAuthenticationOAuth oauth = (KafkaClientAuthenticationOAuth) authentication;
+
+                    if (oauth.getTlsTrustedCertificates() != null && !oauth.getTlsTrustedCertificates().isEmpty()) {
+                        hasClusterOauthTrustedCerts = true;
+                    }
+                    appendClusterOAuthSecretSource(clustersOauthClientSecrets, clusterAlias, oauth.getClientSecret());
+                    appendClusterOAuthSecretSource(clustersOauthAccessTokens, clusterAlias, oauth.getAccessToken());
+                    appendClusterOAuthSecretSource(clustersOauthRefreshTokens, clusterAlias, oauth.getRefreshToken());
+                }
+            }
+        }
+
+        varList.add(buildEnvVar(ENV_VAR_KAFKA_MIRRORMAKER_2_CLUSTERS, clusterAliases.toString()));
+
+        if (clustersTrustedCerts.length() > 0) {
```

Comment:
`isEmpty()`

Reply:
This `clustersTrustedCerts` object is a StringBuilder, which doesn't have `isEmpty()` :(

Reply:
Excellent point :-)